woocommerce/woocommerce
# Add functionality to add multiple items and apply coupon from URL

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This pull request adds functionality to allow multiple items to be added to the cart and apply a coupon code from the URL parameters using the `format=multi` parameter. This extends WooCommerce's existing format with minimal disruption. A "Format" flag ensures that the legacy behavior remains unchanged while introducing new functionality.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure you have the latest version of WooCommerce installed and activated.
2. Apply the changes from this pull request to your WooCommerce installation.
3. Open your browser and navigate to a URL with the following format:
   ```
   http://localhost:8888/?add-to-cart=15,13&quantity=1,1&coupon-code=free20&format=multi
   ```
4. Verify that products with IDs 15 and 13 are added to the cart with the specified quantities.
5. Verify that the coupon code `free20` is applied to the cart.
6. Check that the legacy behavior (e.g., `http://localhost:8888/?add-to-cart=15`) still works as expected.
7. Ensure there are no errors or warnings in the browser console or server logs.

### Changelog entry



<details>

<summary>Changelog Entry Details</summary>

#### Significance


#### Type


#### Message

Add functionality to allow multiple items to be added to the cart and apply a coupon code from the URL parameters using the `format=meta` parameter.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

This Pull Request does not require a changelog entry because it introduces a new feature that does not affect existing functionality.

</details>

### Screenshots



<img width="1721" alt="Screenshot 2024-08-21 at 3 25 53 PM" src="https://github.com/user-attachments/assets/4cfa29b5-2687-405d-a31a-24b616f6a066">
<img width="442" alt="Screenshot 2024-08-21 at 3 36 41 PM" src="https://github.com/user-attachments/assets/9d79211b-8e68-4271-b9c3-78db0f3e1472">
